File: hyperledger/cpp/configure/configure/network_deploy.cpp
#include "network_deploy.h"
#include <iostream>
#include <jsoncpp/json/json.h>
#include <fstream>

namespace config {

bool Network_deploy::JsonLoadfile(std::string path)
{
	std::ifstream stream;
	stream.open(path);

	if (stream.fail()) {
		printf("file open fail\n");
		return false;
	}
	
	Json::Value root;
	stream >> root;

	return Serialize(root);
}

bool Network_deploy::Serialize(Json::Value value)
{
	if (value.empty())
		return false;

	Json::Value peers, orderers;

	peers = value["peers"];
	orderers = value["orderers"];

	Peer peer;
	Orderer orderer;
	for (int i = 0; i < peers.size(); i++)
	{
		peer.id = peers[i]["id"].asString();
		peer.ip = peers[i]["ip"].asString();
		peer.admin_listen = peers[i]["admin_listen"].asString();
		peer.endorsor_listen = peers[i]["endorsor_listen"].asString();
		peer.chaincode_listen = peers[i]["chaincode_listen"].asString();

		_peers.push_back(peer);
	}

	for (int i = 0; i < orderers.size(); i++)
	{
		orderer.id = orderers[i]["id"].asString();
		orderer.ip = orderers[i]["ip"].asString();
		orderer.listen = orderers[i]["listen"].asString();

		_orderers.push_back(orderer);
	}

	return true;
}

void Network_deploy::Initialize()
{
	_peers.clear();
	_orderers.clear();
}

}

File: hyperledger/go/test/network_deploy.go
package config

import (
	"fmt"
	"os"
	"io/ioutil"
)

type Network_deploy struct{
	Peers []Peer `json:"peers"`
	Orderers []Orderer `json:"orderers"`
}

type Peer struct{
	ID string `json:"id"`
	IP string `json:"ip"`
	IP2 string `json:"ip2"`
	Admin_listen string `json:"admin_listen"`
	Endorsor_listen string `json:"endorsor_listen"`
	Chaincode_listen string `json:"chaincode_listen"`
}

type Orderer struct{
	ID string `json:"id"`
	IP string `json:"ip"`
	Listen string `json:"listen"`
}

func JsonLoadFile(Path string)[]byte{
	jsonFile, err := os.Open(Path)

	if err != nil{
		fmt.Println(err)
	}

	byteValue, _ := ioutil.ReadAll(jsonFile)
	
	return byteValue

}
